corehq/apps/cowin/repeaters.py

```diff
@@ -0,0 +1,73 @@
+from django.utils.translation import ugettext_lazy as _
+
+from corehq import toggles
+from corehq.apps.cowin.repeater_generators import (
+    BeneficiaryRegistrationPayloadGenerator,
+    BeneficiaryVaccinationPayloadGenerator,
+)
+from corehq.apps.hqcase.utils import update_case
+from corehq.motech.repeaters.models import CaseRepeater
+
+
+class BaseCOWINRepeater(CaseRepeater):
+    class Meta:
+        app_label = 'repeaters'
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return toggles.COWIN_INTEGRATION.enabled(domain)
+
+    def get_headers(self, repeat_record):
+        headers = super(BaseCOWINRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'Accept-Language': 'en_US',
+            'User-Agent': '',
+            'Accept': 'application/json',
+            'Content-Type': 'application/json',
+        })
+        return headers
+
+
+class BeneficiaryRegistrationRepeater(BaseCOWINRepeater):
+    payload_generator_classes = (BeneficiaryRegistrationPayloadGenerator,)
+    friendly_name = _("Register beneficiaries on COWIN")
+
+    def get_headers(self, repeat_record):
+        headers = super(BeneficiaryRegistrationRepeater, self).get_headers(repeat_record)
+        headers.update({
+            'X-Api-Key': self.connection_settings.plaintext_password,
+        })
```

Comment:
It looks like this is repeated in `BeneficiaryVaccinationRepeater`, and could be moved to `BaseCOWINRepeater`.

Reply:
Right, will move it there.

It's not necessary that all api end points need the api key but the repeaters we have added so far, need it. Knowledge of other non-api-key end points clouded my judgement. Thanks for pointing that out.

Reply:
https://github.com/dimagi/commcare-hq/pull/29949/commits/d5465951dad4134c5568b4d5187f1f7612c08450